src/conn/conn_evict.c

```diff
@@ -0,0 +1,356 @@
+    WT_RET(__wti_evict_config(session, cfg, false));
+
+    /*
+     * The lowest possible page read-generation has a special meaning, it marks a page for forcible
+     * eviction; don't let it happen by accident.
+     */
+    cache->read_gen_oldest = WT_READGEN_START_VALUE;
+    __wt_atomic_store64(&cache->read_gen, WT_READGEN_START_VALUE);
+
+    WT_RET(__wt_cond_auto_alloc(
+      session, "cache eviction server", 10 * WT_THOUSAND, WT_MILLION, &cache->evict_cond));
+    WT_RET(__wt_spin_init(session, &cache->evict_pass_lock, "evict pass"));
+    WT_RET(__wt_spin_init(session, &cache->evict_queue_lock, "cache eviction queue"));
+    WT_RET(__wt_spin_init(session, &cache->evict_walk_lock, "cache walk"));
+    if ((ret = __wt_open_internal_session(
+           conn, "evict pass", false, WT_SESSION_NO_DATA_HANDLES, 0, &cache->walk_session)) != 0)
+        WT_RET_MSG(NULL, ret, "Failed to create session for eviction walks");
+
+    /* Allocate the LRU eviction queue. */
+    cache->evict_slots = WT_EVICT_WALK_BASE + WT_EVICT_WALK_INCR;
+    for (i = 0; i < WT_EVICT_QUEUE_MAX; ++i) {
+        WT_RET(__wt_calloc_def(session, cache->evict_slots, &cache->evict_queues[i].evict_queue));
+        WT_RET(__wt_spin_init(session, &cache->evict_queues[i].evict_lock, "cache eviction"));
+    }
+
+    /* Ensure there are always non-NULL queues. */
+    cache->evict_current_queue = cache->evict_fill_queue = &cache->evict_queues[0];
+    cache->evict_other_queue = &cache->evict_queues[1];
+    cache->evict_urgent_queue = &cache->evict_queues[WT_EVICT_URGENT_QUEUE];
+
+    /*
+     * We get/set some values in the evict statistics (rather than have two copies), configure them.
+     */
+    __wti_evict_stats_update(session);
+    return (0);
+}
+
+/*
+ * __wti_evict_destroy --
+ *     Destroy Eviction
+ */
+int
+__wti_evict_destroy(WT_SESSION_IMPL *session)
+{
+    WT_CACHE *cache;
+    WT_DECL_RET;
+    int i;
+
+    cache = S2C(session)->cache;
+
+    if (cache == NULL)
+        return (0);
+
+    __wt_cond_destroy(session, &cache->evict_cond);
+    __wt_spin_destroy(session, &cache->evict_pass_lock);
+    __wt_spin_destroy(session, &cache->evict_queue_lock);
+    __wt_spin_destroy(session, &cache->evict_walk_lock);
+    if (cache->walk_session != NULL)
+        WT_TRET(__wt_session_close_internal(cache->walk_session));
+
+    for (i = 0; i < WT_EVICT_QUEUE_MAX; ++i) {
+        __wt_spin_destroy(session, &cache->evict_queues[i].evict_lock);
+        __wt_free(session, cache->evict_queues[i].evict_queue);
+    }
+
+    return (ret);
+}
+
+/*
+ * __wti_evict_stats_update --
+ *     Update the eviction statistics for return to the application.
+ */
+void
+__wti_evict_stats_update(WT_SESSION_IMPL *session)
+{
+    WT_CACHE *cache;
+    WT_CONNECTION_IMPL *conn;
+    WT_CONNECTION_STATS **stats;
+
+    conn = S2C(session);
+    cache = conn->cache;
+    stats = conn->stats;
+
+    WT_STATP_CONN_SET(session, stats, cache_eviction_maximum_page_size,
```

Comment:
What are your plans for changing the stat names (`cache_eviction_maximum_page_size`) here? It seems weird to retain the `cache_` prefix.
T2 stats are sorted on the description string so the name changes should have no external impact.

Reply:
Yes, names would be changes in the struct split ticket. Rn, fields are still part of the cache and hence, I didn't change the name.